### PR TITLE
(CDAP-3713) Removed all catch blocks in HTTP Handlers where we respon…

### PR DESCRIPTION
**Pull request by bdmogal**
…d with an empty message, and let HttpExceptionHandler respond with a more descriptive message instead. This would help immensely while debugging test failures.

Jira: [CDAP-3713](https://issues.cask.co/browse/CDAP-3713)
Build: http://builds.cask.co/browse/CDAP-RBT453-1

### CODE REVIEW

File: cdap-app-fabric/src/main/java/co/cask/cdap/gateway/handlers/ProgramLifecycleHttpHandler.java [hunk 1]
```diff
@@ -490,18 +481,13 @@ public void getProgramRuntimeArgs(HttpRequest request, HttpResponder responder,
...
-        return;
-      }
-      Map<String, String> runtimeArgs = preferencesStore.getProperties(id.getNamespaceId(), appId,
-                                                                       programType, programId);
-      responder.sendJson(HttpResponseStatus.OK, runtimeArgs);
-    } catch (Throwable e) {
-      LOG.error("Error getting runtime args {}", e.getMessage(), e);
-      responder.sendStatus(HttpResponseStatus.INTERNAL_SERVER_ERROR);
+    if (!store.programExists(id)) {
+      responder.sendString(HttpResponseStatus.NOT_FOUND, "Program not found");
```
Comment (sagarkapare) [inline cdap-app-fabric/src/main/java/co/cask/cdap/gateway/handlers/ProgramLifecycleHttpHandler.java]: throw NotFoundException here?

File: cdap-app-fabric/src/main/java/co/cask/cdap/gateway/handlers/ProgramLifecycleHttpHandler.java [hunk 2]
```diff
@@ -522,18 +508,13 @@ public void saveProgramRuntimeArgs(HttpRequest request, HttpResponder responder,
...
-        return;
-      }
-      Map<String, String> args = decodeArguments(request);
-      preferencesStore.setProperties(namespaceId, appId, programType, programId, args);
-      responder.sendStatus(HttpResponseStatus.OK);
-    } catch (Throwable e) {
-      LOG.error("Error getting runtime args {}", e.getMessage(), e);
-      responder.sendStatus(HttpResponseStatus.INTERNAL_SERVER_ERROR);
+    if (!store.programExists(id)) {
+      responder.sendString(HttpResponseStatus.NOT_FOUND, "Program not found");
```
Comment (sagarkapare) [inline cdap-app-fabric/src/main/java/co/cask/cdap/gateway/handlers/ProgramLifecycleHttpHandler.java]: same here

File: cdap-app-fabric/src/main/java/co/cask/cdap/gateway/handlers/TransactionHttpHandler.java [hunk 3]
```diff
@@ -115,66 +114,52 @@ public void invalidateTx(HttpRequest request, HttpResponder responder,
...
-        return;
-      }
-
-      if (body == null || !body.containsKey("time")) {
-        responder.sendString(HttpResponseStatus.BAD_REQUEST, "Time not specified");
-        return;
-      }
+      body = parseBody(request, STRING_LONG_MAP_TYPE);
+    } catch (IllegalArgumentException e) {
+      responder.sendString(HttpResponseStatus.BAD_REQUEST, "Invalid time value in request");
```
Comment (sagarkapare) [inline cdap-app-fabric/src/main/java/co/cask/cdap/gateway/handlers/TransactionHttpHandler.java]: throw BadRequestException here

File: cdap-app-fabric/src/main/java/co/cask/cdap/gateway/handlers/TransactionHttpHandler.java [hunk 4]
```diff
@@ -115,66 +114,52 @@ public void invalidateTx(HttpRequest request, HttpResponder responder,
...
+    }
 
-      long time = body.get("time");
-      txClient.truncateInvalidTxBefore(time);
-      responder.sendStatus(HttpResponseStatus.OK);
-    } catch (Exception e) {
-      LOG.error("Got exception: ", e);
-      responder.sendStatus(HttpResponseStatus.INTERNAL_SERVER_ERROR);
+    if (body == null || !body.containsKey("time")) {
+      responder.sendString(HttpResponseStatus.BAD_REQUEST, "Time not specified");
```
Comment (sagarkapare) [inline cdap-app-fabric/src/main/java/co/cask/cdap/gateway/handlers/TransactionHttpHandler.java]: same

File: cdap-app-fabric/src/main/java/co/cask/cdap/gateway/handlers/ProgramLifecycleHttpHandler.java (hunk 1 above)
Comment (bdmogal) [inline cdap-app-fabric/src/main/java/co/cask/cdap/gateway/handlers/ProgramLifecycleHttpHandler.java]: I don't want to change any non-``INTERNAL_SERVER_ERROR`` responses in this PR. Is that ok?

File: cdap-app-fabric/src/main/java/co/cask/cdap/gateway/handlers/TransactionHttpHandler.java [hunk 5]
```diff
@@ -115,66 +114,52 @@ public void invalidateTx(HttpRequest request, HttpResponder responder,
...
-        return;
-      }
-
-      if (body == null || !body.containsKey("ids")) {
-        responder.sendString(HttpResponseStatus.BAD_REQUEST, "Transaction ids not specified");
-        return;
-      }
+      body = parseBody(request, STRING_LONG_SET_MAP_TYPE);
+    } catch (IllegalArgumentException e) {
+      responder.sendString(HttpResponseStatus.BAD_REQUEST, "Invalid ids specified in request");
```
Comment (sagarkapare) [inline cdap-app-fabric/src/main/java/co/cask/cdap/gateway/handlers/TransactionHttpHandler.java]: same here

File: cdap-app-fabric/src/main/java/co/cask/cdap/gateway/handlers/TransactionHttpHandler.java [hunk 6]
```diff
@@ -115,66 +114,52 @@ public void invalidateTx(HttpRequest request, HttpResponder responder,
...
+    }
 
-      Set<Long> txIds = body.get("ids");
-      txClient.truncateInvalidTx(txIds);
-      responder.sendStatus(HttpResponseStatus.OK);
-    } catch (Exception e) {
-      LOG.error("Got exception: ", e);
-      responder.sendStatus(HttpResponseStatus.INTERNAL_SERVER_ERROR);
+    if (body == null || !body.containsKey("ids")) {
+      responder.sendString(HttpResponseStatus.BAD_REQUEST, "Transaction ids not specified");
```
Comment (sagarkapare) [inline cdap-app-fabric/src/main/java/co/cask/cdap/gateway/handlers/TransactionHttpHandler.java]: here as well.

File: cdap-app-fabric/src/main/java/co/cask/cdap/gateway/handlers/ProgramLifecycleHttpHandler.java (hunk 1 above)
Comment (sagarkapare) [inline cdap-app-fabric/src/main/java/co/cask/cdap/gateway/handlers/ProgramLifecycleHttpHandler.java]: sounds good.

**Comment by sagarkapare** [comment]
LGTM. :+1:

File: cdap-app-fabric/src/main/java/co/cask/cdap/gateway/handlers/ProgramLifecycleHttpHandler.java (hunk 2 above)
Comment (bdmogal) [inline cdap-app-fabric/src/main/java/co/cask/cdap/gateway/handlers/ProgramLifecycleHttpHandler.java]: ditto

File: cdap-app-fabric/src/main/java/co/cask/cdap/gateway/handlers/TransactionHttpHandler.java (hunk 3 above)
Comment (bdmogal) [inline cdap-app-fabric/src/main/java/co/cask/cdap/gateway/handlers/TransactionHttpHandler.java]: ditto

**Comment by bdmogal** [comment]
We can do other exceptions in future. I specifically opened this for those ``INTERNAL_SERVER_ERROR``s which had an empty response.

Build is green. Merging.

**Pull request merged**